### DISCUSSION TITLE
Print From USB?

### DISCUSSION THREAD

**Discussion by Zeltroix** [OrcaSlicer/OrcaSlicer#1288] [Q&A]
Is there a print from USB option like in Cura? Using an ender 3 v2 neo.

**Comment by GrahamDyer** [comment]
Bump

Yes, if there isn't, why not? It's such a convenient way to print.

**Reply by foreachthing** [reply]
I could print via USB. Tried it a few times. Very annoying feature IMHO. If cura starts it resets USB. So, one print printing (this was started from SD), start 2nd Cura, print job fails because of reset. I'm done. I don't miss this feature.
https://github.com/Ultimaker/Cura/issues/1478

**Reply by vontrapp** [reply]
Then don't use the feature. No need to take it from the rest of us.

**Reply by akadata** [reply]
> Then don't use the feature. No need to take it from the rest of us.

Funny reply, I cant print to USB either, sold on the interface, sold on the features, lost on the lack of serial port support!

**Reply by KingSidorak** [reply]
@foreachthing I ran a 10x Ender 3 farm for around 2 years using Cura over USB without any issues that were not my fault, or Windows Driver issues. This is completely feasible if the hardware and drivers are good.

**Comment by ShaGGyUK** [comment]
would bring many cura users over if usb was supported.

**Reply by Beanlipe** [reply]
yes! everything so I dont have to manage sd cards

**Reply by akadata** [reply]
I can not understand the lack of support for USB, the printer has a USB port for a reason!

**Reply by otede** [reply]
> would bring many cura users over if usb was supported.

We'd better get that before Bambu actually adds support for Cura. Then we might see some real influx back to Cura due to it's interface superiority (mainly due to the driving-driven settings philosophy).

**Reply by ilko-k** [reply]
This is subjective. Personally I still prefer Cura and the only thing that made me test and use Orca is the scarf seam feature. If Cura ever includes that, it will be a game changer because Orca inherited all of the downsides of Prusa slicer, notably the so annoying and well known dependency system of presets.

**Reply by otede** [reply]
> This is subjective. Personally I still prefer Cura and the only thing that made me test and use Orca is the scarf seam feature. If Cura ever includes that, it will be a game changer because Orca inherited all of the downsides of Prusa slicer, notably the so annoying and well known dependency system of presets.

Could you please link an article explaing differences between Cura vs Orca preset dependency systems?

**Comment by ShaGGyUK** [comment]
So if that's the case why didn't you do the obvious thing and UNPLUG the USB that you don't use?

Simples.

Many people do use it.



Get BlueMail for Android

**Comment by merqlove** [comment]
looks like here is misconfiguration for usb
i c usage of [this one ](https://github.com/SoftFever/OrcaSlicer/blob/e4255b3c01fd6fe3b987ce38235739700be8e411/src/hidapi/README.md?plain=1#L3) from bamboolab, but looks likes it works not always

**Comment by vontrapp** [comment]
This used to be a feature, and now it's not?

WHY??

Bring it back.

**Reply by paulnaude** [reply]
When you printed via USB, did the printer show the time remaining etc., or did Creality Print show it, Like Cura does?
With Cura, printing to my Ender 3 v3 SE, the printer displays nothing (just the menu) while Cura shows the progress.

**Reply by KingSidorak** [reply]
I just got an email for this, so I am not sure who this is to; but when I printed via USB on Ender 3s I was using Cura. It would not only show time remaining, but update the time remaining depending on how slow the machine was being

**Comment by ilko-k** [comment]
+1

**Comment by twintersx** [comment]
USB is a must!

**Reply by srt-jay** [reply]
It's basic functionality, even. Imagine a router with no Ethernet ports (AHEM Starlink!)

**Comment by hellimi** [comment]
I don't know if it gives the same result, but I save the gcode from Orca and import it into Cura and print to USB from there... Not ideal but so far it works.

**Reply by Gean97** [reply]
I do the same thing, but I import it into Pronterface. I highly recommend it. Pronterface gives me better results than Cura when importing G-code. It also has better printer control and a better code terminal (it shows everything you send and all the printer responses), and it lets you view and edit the G-code you import. It shows you exactly what line of code you're on and even shows an animation of the printing progress.

**Comment by srt-jay** [comment]
I can't believe there still isn't a solution for this that doesn't involve user contortion and installing several pieces of software.
How can this be considered mature software if it can't even support USB??? There is literally NO REASON for it not to after this many years.

Maybe the devs got lucky and aren't that gifted afterall.

**Reply by pgifford** [reply]
Ask for a refund, I bet they'll give you double your money back.

**Comment by paulnaude** [comment]
Very strange. I was thinking the whole time that I was being stupid or that
the printer was defective. Never thought it would be a limitation with
OrcaSlicer

Paul Naudé

**Comment by ilko-k** [comment]
For people really missing USB prints with Orca, Octoprint is the solution ("localhost" configuration)
